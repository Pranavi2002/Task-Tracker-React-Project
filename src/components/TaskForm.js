import React, { useState } from "react";
import { Button, TextField, Stack } from "@mui/material";
import { motion } from "framer-motion";

function TaskForm({ onAddTask }) {
  const [input, setInput] = useState("");
  const [dueDate, setDueDate] = useState("");
  const [errors, setErrors] = useState({});

  const handleSubmit = (e) => {
    e.preventDefault();
    const todayStr = new Date().toISOString().split("T")[0];
    const validationErrors = {};
    if (!input.trim()) validationErrors.input = "Task name is required";
    if (dueDate && dueDate < todayStr) validationErrors.dueDate = "Due date cannot be in the past";

    if (Object.keys(validationErrors).length > 0) {
      setErrors(validationErrors);
      return;
    }

    onAddTask(input, dueDate || "");
    setInput("");
    setDueDate("");
    setErrors({});
  };

  return (
    <form onSubmit={handleSubmit}>
      <Stack direction={{ xs: "column", sm: "row" }} spacing={2} sx={{ mt: 2, alignItems: "center" }}>
        <TextField
          label="Task Name"
          value={input}
          onChange={(e) => setInput(e.target.value)}
          error={!!errors.input}
          helperText={errors.input || ""}
          fullWidth
        />
        <TextField
          label="Due Date"
          type="date"
          value={dueDate}
          onChange={(e) => setDueDate(e.target.value)}
          InputLabelProps={{ shrink: true }}
          error={!!errors.dueDate}
          helperText={errors.dueDate || ""}
        />
        <motion.div whileHover={{ scale: 1.05 }} whileTap={{ scale: 0.95 }}>
          <Button type="submit" variant="contained" color="primary">
            Add
          </Button>
        </motion.div>
      </Stack>
    </form>
  );
}

export default TaskForm;