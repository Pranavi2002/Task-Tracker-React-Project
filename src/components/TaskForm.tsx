import React, { useState, FormEvent } from "react";
import { motion } from "framer-motion";

interface TaskFormProps {
  onAddTask: (text: string, dueDate: string) => void;
}

interface Errors {
  input?: string;
  dueDate?: string;
}

const TaskForm: React.FC<TaskFormProps> = ({ onAddTask }) => {
  const [input, setInput] = useState("");
  const [dueDate, setDueDate] = useState("");
  const [errors, setErrors] = useState<Errors>({});

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();

    const validationErrors: Errors = {};

    if (!input.trim()) {
      validationErrors.input = "Task name is required";
    }

    const today = new Date();
    const yyyy = today.getFullYear();
    const mm = String(today.getMonth() + 1).padStart(2, "0");
    const dd = String(today.getDate()).padStart(2, "0");
    const todayStr = `${yyyy}-${mm}-${dd}`;

    if (dueDate && dueDate < todayStr) {
      validationErrors.dueDate = "Due date cannot be in the past";
    }

    if (Object.keys(validationErrors).length > 0) {
      setErrors(validationErrors);
      return;
    }

    onAddTask(input, dueDate || "");
    setInput("");
    setDueDate("");
    setErrors({});
  };

  const inputStyle: React.CSSProperties = {
    padding: "8px",
    borderRadius: "5px",
    border: "1px solid #ccc",
    marginRight: "10px",
  };

  const addButtonStyle: React.CSSProperties = {
    padding: "6px 12px",
    border: "none",
    borderRadius: "5px",
    cursor: "pointer",
    backgroundColor: "#4caf50",
    color: "#fff",
    transition: "0.3s",
  };

  return (
    <form onSubmit={handleSubmit} style={{ marginTop: "20px" }}>
      <input
        type="text"
        placeholder="Enter a task"
        value={input}
        onChange={(e) => setInput(e.target.value)}
        style={inputStyle}
      />
      {errors.input && <div style={{ color: "red" }}>{errors.input}</div>}

      <input
        type="date"
        value={dueDate}
        onChange={(e) => setDueDate(e.target.value)}
        style={inputStyle}
      />
      {errors.dueDate && <div style={{ color: "red" }}>{errors.dueDate}</div>}

      <motion.button
        type="submit"
        style={addButtonStyle}
        whileHover={{ scale: 1.05 }}
        whileTap={{ scale: 0.95 }}
      >
        Add
      </motion.button>
    </form>
  );
};

export default TaskForm;