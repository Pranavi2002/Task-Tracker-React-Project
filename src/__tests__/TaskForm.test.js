// src/__tests__/TaskForm.test.js
import React from "react";
import { render } from "@testing-library/react";
import TaskForm from "../components/TaskForm";

describe("TaskForm Snapshot", () => {
  // Mock function to simulate adding a task
  const onAddTask = jest.fn();

  it("renders correctly", () => {
    // Render TaskForm component
    const { asFragment } = render(<TaskForm onAddTask={onAddTask} />);
    
    // Take a snapshot to catch UI changes in the future
    expect(asFragment()).toMatchSnapshot();
  });
});