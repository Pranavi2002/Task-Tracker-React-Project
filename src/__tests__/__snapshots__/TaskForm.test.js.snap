// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm Snapshot renders correctly 1`] = `
<DocumentFragment>
  <form
    style="margin-top: 20px;"
  >
    <input
      placeholder="Enter a task"
      style="padding: 8px; border-radius: 5px; border: 1px solid #ccc; margin-right: 10px;"
      type="text"
      value=""
    />
    <input
      style="padding: 8px; border-radius: 5px; border: 1px solid #ccc; margin-right: 10px;"
      type="date"
      value=""
    />
    <button
      style="padding: 6px 12px; border-radius: 5px; cursor: pointer; background-color: rgb(76, 175, 80); color: rgb(255, 255, 255); transition: 0.3s;"
      tabindex="0"
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
